src/space.js

```diff
@@ -263,20 +268,46 @@ export default class ColorSpace {
 		return [...new Set(Object.values(ColorSpace.registry))];
 	}
 
+	static addToRegistry (id, space) {
+		if (this.registry[id] && this.registry[id] !== space) {
+			throw new Error(`Duplicate color space registration: '${id}'`);
+		}
+		this.registry[id] = space;
+	}
+
+	static validateIds () {
+		// Get full list of IDs.
+		const ids = this.all.flatMap(space =>
+			[space.id, ...(space.formats.color.ids ?? [])]
+		).sort();
+
+		let prev;
+		for (let id of ids) {
+			if (id === prev) {
+				throw new Error(`Duplicate color ID registration: '${id}'`);
+			}
+			prev = id;
+		}
+	}
+
 	static register (id, space) {
 		if (arguments.length === 1) {
 			space = arguments[0];
 			id = space.id;
 		}
 
 		space = this.get(space);
-		this.registry[id] = space;
-		return space;
-	}
+		this.addToRegistry(id, space);
```

Comment:
Do we need a separate method? When somebody is using the `register(id, space)` signature, we should not be registering any additional things like aliases. So all we need to do is preserve what signature was used. Then we can recursively call it to register a color space with each alias, and still only have the checking code in one place.

Reply:
Sorry, I'm not entirely sure I understand 😕

Do you mean that we should't have the aliases as part of the color space definition (`space.aliases`), and we should instead delegate the responsibility of registering the alias to the consumer? Or do you mean that we should simply call `register` recursively from within itself, but keep the responsibility of registering aliases within the `ColorSpace` class?

Reply:
I mean, there is a semantic difference between the two signatures here. If someone calls `ColorSpace.register("foobar", mySpace)` they intend to register `mySpace` under the `"foobar"` alias, and not do any other registration. E.g. someone may import a color space and want to register it with a different id altogether. If they do `ColorSpace.register(mySpace)`, then it makes sense to register it with its id and aliases.

This also means that *we* can use it that way internally: When `ColorSpace.register()` is used with a single argument, we can internally call `ColorSpace.register(id, space)` as well as `ColorSpace.register(alias, space)` for every alias, thus not needing a separate function for this.

Does it make sense now?

Reply:
Oh, I see, thank you for clarifying that! Yes, that sounds like a good approach 👍

Reply:
Done!